### ISSUE TITLE
Bug in liveness-violation report  

### ISSUE THREAD

**Issue by db7** [MPI-SWS/genmc#34]
GenMC 0.7 has a small bug in the reporting of liveness violations. It seems to always point to thread 1.

Here is a simple example:

```c
#include <pthread.h>

int x,y;

void* run(void *arg) {
	if (y) 
		x = 1;
	return 0;
}

int main()
{
	pthread_t t;
	pthread_create(&t, 0, run, 0);
	y = 1;
	while (!x);  // thread 0 (main thread) may hang here
	pthread_join(t, 0);
	return 0;
}
``` 

With GenMC 0.7 I get:

```
$ genmc -check-liveness -mo -imm bug-liveness.c 
Error detected: Liveness violation!
Event (1, 2) in graph:
<-1, 0> main:
        (0, 0): B
        (0, 1): TC [forks 1] L.14
        (0, 2): Wna (y, 1) L.16
        (0, 3): LOOP_BEGIN
        (0, 4): SPIN_START
        (0, 5): Rna (x, 0) [INIT] L.17
<0, 1> run:
        (1, 0): B
        (1, 1): Rna (y, 0) [INIT] L.6
        (1, 2): E

Non-terminating spinloop: thread 1     <==== HERE
Number of complete executions explored: 1
Number of blocked executions seen: 2
Total wall-clock time: 0.02s
```

But the thread that hangs is thread 0.

With GenMC 0.6.1 the report was correct:

```
$ genmc -check-liveness -mo -imm  bug-liveness.c 
Error detected: Liveness violation!
Event (1, 2) in graph:
<-1, 0> main:
        (0, 0): B
        (0, 1): TC [forks 1] L.14
        (0, 2): Wna (y, 1) L.16
        (0, 3): LOOP_BEGIN
        (0, 4): SPIN_START
        (0, 5): Rna (x, 0) [INIT] L.17
<0, 1> run:
        (1, 0): B
        (1, 1): Rna (y, 0) [INIT] L.6
        (1, 2): E

Non-terminating spinloop: thread 0
Number of complete executions explored: 0
Number of blocked executions seen: 1
Total wall-clock time: 0.02s
```

I found also a bit confusing what this means: `Event (1, 2) in graph`. Is this correct?

**Comment by michaliskok** [comment]
Thanks for the report! I will push a fix in the next release (hopefully before mid-February).

> I found also a bit confusing what this means: `Event (1, 2) in graph`. Is this correct?

No, it should print (0, 5). (It prints the last event of the thread it considers stuck.)

Michalis

**Comment by michaliskok** [comment]
Fixed in v0.8.
